crates/subspace-service/src/dsn/import_blocks.rs

```diff
@@ -81,8 +87,16 @@ where
     B: BlockT,
     IQ: ImportQueue<B> + 'static,
 {
-    let piece_provider = PieceProvider::<NoPieceValidator>::new(node.clone(), None, false);
+    let piece_provider = PieceProvider::<RecordsRootPieceValidator>::new(
+        node.clone(),
+        Some(RecordsRootPieceValidator::new(
+            node.clone(),
+            Kzg::new(test_public_parameters()),
```

Comment:
KZG must be an argument here too, ideally it only needs to be created once during lifetime of the application and passed everywhere else as an argument. This is especially true when we are talking about library code like here where we might want to use other parameters for testing purposes.

Reply:
I'm not sure I follow here. The farmer has a similar usage (local variable in a middle-level function). Let's have a second usage or proper public parameters and decide on the variable place after that.